mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,170 @@
+
+    scale_factor_vectors = _get_scale_factor_vectors(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    num_layers = len(scale_factor_vectors[0])
+
+    # Evaluate the monomial terms using the values in the scale factor vectors
+    # and insert in the sample matrix
+    # each row is specific to each scale factor vector
+    # each column is a term in the monomial basis
+    variable_exp = _full_monomial_basis_term_exponents(num_layers, degree)
+    sample_matrix = np.empty((len(variable_exp), len(variable_exp)))
+
+    for i, scale_factors in enumerate(scale_factor_vectors):
+        for j, exponent in enumerate(variable_exp):
+            evaluated_terms = []
+            for base, exp in zip(scale_factors, exponent):
+                # raise scale factor value by the exponent dict value
+                evaluated_terms.append(base**exp)
+            # multiply both elements in the list to create an evaluated
+            # monomial term
+            sample_matrix[i, j] = np.prod(evaluated_terms)
+
+    return sample_matrix
+
+
+def linear_combination_coefficients(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+) -> list[float]:
+    r"""
+    Defines the function to find the linear combination coefficients from the
+    sample matrix as required for multivariate extrapolation (defined in
+    :cite:`Russo_2024_LRE`).
+
+    Args:
+        input_circuit: Circuit to be scaled.
+        degree: Degree of the multivariate polynomial.
+        fold_multiplier: Scaling gap required by unitary folding.
+        num_chunks: Number of desired approximately equal chunks. When the
+            number of chunks is the same as the layers in the input circuit,
+            the input circuit is unchanged.
+
+    Returns:
+        List of the evaluated monomial basis terms using the scale factor
+            vectors.
+    """
+    input_sample_matrix = sample_matrix(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    try:
+        det = np.linalg.det(input_sample_matrix)
+    except RuntimeWarning:  # pragma: no cover
+        # taken from https://stackoverflow.com/a/19317237
+        warnings.warn(  # pragma: no cover
+            "To account for overflow error, required determinant of "
+            + "large sample matrix is calculated through "
+            + "`np.linalg.slogdet`."
+        )
+        sign, logdet = np.linalg.slogdet(  # pragma: no cover
+            input_sample_matrix
+        )
+        det = sign * np.exp(logdet)  # pragma: no cover
+
+    if np.isinf(det):
+        raise ValueError(  # pragma: no cover
+            "Determinant of sample matrix cannot be calculated as "
+            + "the matrix is too large. Consider chunking your"
+            + " input circuit. "
+        )
+    assert det != 0.0
+
+    coeff_list = []
+    mat_row, mat_cols = input_sample_matrix.shape
+    assert mat_row == mat_cols
```

Comment:
Similar question here as to whether we want to provide more details to the end-user as to what the issue would have been here if this assert fails.

Reply:
See https://github.com/unitaryfund/mitiq/pull/2447#discussion_r1729114577